# python basics



1. what is python and why it is widely used in data analytics ? mention any three reason .
- python is a high-level , easy-to-learn programming language used to write instruction for computers . it is knowns for its simple syntax (almost like english) , which makes it beginner-friendly and powerful for many tasks like web development , automation , and especially data analytics .

python is widely used in data analytics because it makes working with data simple , fast , and powerful .python helps create graphs , charts , and dashbords which make data easy to understand and present .

here are 3 main reason -

# simple and easy to understand
python has a clean , readable  syntax (almost like english ), so you don't need to spend too much time learning complex rules. this helps analysts focus more on analyzing data instead of coding .

# Powerful libraries
python has strong libraries that make data work easier:

Nummpy- for numerical calculation
Pandas - for data cleaning and analysis
Matplotlib & Seaborn - for data visualization

These libraries save time and reduce effort .

# Handles large data
python can easily work with large datasets and integrates with big data tools .
this makes it useful in real-world business analytics .

2. Explain the difference between List and Tuple in Python?
- In python, both lists and tuples are used to store multiple items in a single variable . however, they have several important differences based on their properties and usage.

# Defination
 A List is a collection of elements that is mutable , meaning its value can be changed after creation .
 A tuple is a collection of elements that is immutable , meaning its values cannot be changed once defined.

# syntax
Lists are created using square brackets []
whereas tuples are created using paretheses()
Example:
. List - [1,2,3]
. Tuple- (1,2,3)

# Mutability
Lists are mutable ,so elements can be added , removed , or modify .
tuples are immitable , so elements remain constant throghout the program .

# Performance
 tuples are geneeally faster than lists because they are fixed and require less memory . Lists are slightly slower due to their dynamic nature .

# Methods
Lists have many buil-in methods such as append (),remove (),and sort().
tuples have limited methods like count() and index().

# Use cases
Lists are used when data needs to be modified frequently .
Tuple are used when data should remain constant and secure .

lists and tuples area both useful data structures in python . Lists provide flexibility due to their mutable nature , while tuples ensure data integrity because they cannot be changed . the choice between them depends on the requirement of the program .

3. What is a function in Python? Why are functions useful?
- A fuction in python is a block of orgnized , reusable code that is used to perform a specific task. instead of writing the same code again and again , we can  define a function once and use or multiple times .A function is an important concept in python that helps make program simple , organized , and efficient .

functions are defined using the keyword def-
Example:
def greet()
    print("hello")

greet() #calling the function

The main uses are:

# 1. code reusability -
 Functions allow you to write code once and use it many times . this saves time and effort .

# 2. Better organization -
Functions help divide a large program into small , manageable parts , making the code easier to understand .

# 3. Avoid Repetition-
instead of writing the same logic again and again , you can call the function whenever needed.

# 4. Easy debugging -
if there is an error , you can check only that function instead of the whole program .

# 5. Improve readability -
Fuctions make code clean and easy to read , especially for others .











4. write a python program to take a user's name as input and print a greeting message?
- a python program to take a user's name as input and print a greeting message


In [9]:
# Take user's name as input
name= input("enter ypur name")

#print greeting message
print("hello,", name + "! welcome")

enter ypur nameankita tomar 
hello, ankita tomar ! welcome


Explanation:
input() function is used to tale the user's name from the keyboard.
The entered name is stored in the variable name.
print() function is used to display the greeting mesage along with the user's name.

output:
Enter your name: ankita tomar
Hello, ankita tomar ! welcome


5. Write
- ### AIM:
To write a python program to check whether a number is even or odd.


In [7]:
num = int(input("enter a number"))

if num % 2 == 0:
  print("even number")
else:
  print("odd number")

enter a number 10
even number


###output:
Enter a number: 10
even number

6.  Write a program to print numbers from 1 to 10 using a loop?
- AIM:
 To write a python program to print numbers form 1 to 10 using a loop.


In [10]:
# program:
## program to print numbers from 1 to 10
for a in range(1,11):
  print(a)

1
2
3
4
5
6
7
8
9
10


explanation:
for loop is used to repeat a block of code.
range (1,11) generates numbers from 1 to 10 .
print(a) prints each number one by one .

7.  Create a list of five numbers and print the maximum number from the list?
- AIM:
 To write a python prohgram to create a list of five numbers and print the maximum number form the list .
  

In [12]:
#program:
#create a list of  five numbers
numbers =[10,25,40,5,15]

#find and print maximum number
print("maximum numer is:",max(numbers))

maximum numer is: 40


Explanation:
A list numbers is created with five values.
max() function is used to find the largest number in the list .
print() displays the result.

Outpt:
maximum number is : 40



8. Write a Python program to remove duplicate values from a list using a set?
- AIM:
To write a python program to remove duplicate values from a list using a set .


In [17]:
#create a list with duplicate values
numbers = [10,20,10,20,40,30,40]

#remove duplicate using set
unique_numbers = list(set(numbers))

#print result
print("list after removing duplicates:",unique_numbers)

list after removing duplicates: [40, 10, 20, 30]


explanation:
A list numbers contains duplicaaate values.
set() automatically removes duplicates elements.
list() converts the set back into a list .
print () displays the final list.

Output:
List after removing duplicates : [40,10,20,30]


9. Write a function that returns the square of a number?
- AIM:
 to write a python function that returns the square of a number .


In [23]:
#function to find square of a number
def square(num):
    return num * num

# take input from user
number = int(input("enter a number: "))

# call function and print result
result = square(number)
print("square is:", result)

enter a number: 25
square is: 625


explanation:
def sqaure(num) : defines a function.
return num*num return the square of the number.
user input is taken using input()
function is called and result is printed .


Output:
Enter a number : 25
sqaure is : 625

10.  Write a Python program to count how many times a number appears in a list.

Example List -[2,3,4,2,5,2]

- To write a python program to count how many times a number appears in a list .


In [24]:
# create a list
numbers = [2,3,4,2,5,2]

#number to be counted
num = 2

# count occurances
count = numbers.count(num)

# print result
print("number appears", count , "times")

number appears 3 times


explNtion:
A list is created with values .
count() function is used to count how many times a number appers .
The result is stored in count
print() displays the output.

OUTPUT:
number appears 3 times